lib/bridge/MatrixHandler.js

```diff
@@ -770,6 +769,83 @@ MatrixHandler.prototype._onMessage = Promise.coroutine(function*(req, event) {
     yield Promise.all(promises);
 });
 
+MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
+    function*(req, ircRoom, ircClient, ircAction, event) {
+
+    // Send the action as is if it is not a text message 
+    //  Also, check for the existance of the getSplitMessages method.
+    if (event.content.msgtype !== "m.text" ||
+        !(ircClient.unsafeClient && ircClient.unsafeClient.getSplitMessages)) {
+        yield this.ircBridge.sendIrcAction(ircRoom, ircClient, ircAction);
+        return;
+    }
+
+    let text = event.content.body;
+
+    // Generate an array of individual messages that would be sent
+    let potentialMessages = ircClient.unsafeClient.getSplitMessages(ircRoom.channel, text);
+    let lineLimit = ircRoom.server.getLineLimit();
+
+    if (potentialMessages.length <= lineLimit) {
+        yield this.ircBridge.sendIrcAction(ircRoom, ircClient, ircAction);
+        return;
+    }
+
+    // Message body too long, upload to HS instead
+    try {
+        // Use the current time as the name of the file
+        //  strip off day of week and timezone
+        //  replace whitespace with underscores
+        // result e.g : Aug_01_2016_13:41:44.txt
+        let fileName = new Date().toString()
+            .match(/.*?\s(.*\:[0-9]*\:[0-9]*).*/)[1]
+            .replace(/\s/g,'_') + ".txt";
```

Comment:
> Note: parsing of date strings with the Date constructor (and Date.parse, they are equivalent) is strongly discouraged due to browser differences and inconsistencies.

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Date#Several_ways_to_create_a_Date_object

There's no guarantees this will work on other Node versions as they are well within their rights to change it. Node fully supports ES5 though, so you *can* use `new Date().toISOString()` and replace from that string representation instead.